### PR TITLE
Add in-progress view to display name EditInPlace

### PR DESCRIPTION
**Pull request by dbkr**
Requires https://github.com/element-hq/compound-web/pull/180

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

File: res/css/views/settings/_UserProfileSettings.pcss [hunk 1]
```diff
@@ -27,6 +27,7 @@ limitations under the License.
         .mx_UserProfileSettings_profile_displayName {
             flex-grow: 1;
             width: 100%;
+            gap: 0;
```
Comment (dbkr) [inline res/css/views/settings/_UserProfileSettings.pcss:30]: Not really related but this is another thing that is specified at the dialog level and we need to override, as it's giving us extra spacing.

File: test/components/views/right_panel/__snapshots__/RoomSummaryCard-test.tsx.snap [hunk 2]
```diff
@@ -123,10 +123,10 @@ exports[`<RoomSummaryCard /> has button to edit topic when expanded 1`] = `
         data-orientation="horizontal"
         role="separator"
       />
-      <label
```
Comment (dbkr) [inline test/components/views/right_panel/__snapshots__/RoomSummaryCard-test.tsx.snap:126]: From https://github.com/element-hq/compound-web/pull/178 in new compound-web

File: playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png
Comment (t3chguy) [inline playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png:1]: The spacing between Display Name and Username looks HUGE, is that intentional?

File: test/components/views/right_panel/__snapshots__/RoomSummaryCard-test.tsx.snap (hunk 2 above)
Comment (t3chguy) [reply test/components/views/right_panel/__snapshots__/RoomSummaryCard-test.tsx.snap:126]: https://github.com/matrix-org/matrix-react-sdk/pull/12586 handles the same for those curious

File: playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png
Comment (dbkr) [reply playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png:1]: ah yes, the bottom margin can go now the remove link is no more

**Review by t3chguy — APPROVED** [review]

**Pull request merged**